sylt-compiler/src/typechecker.rs

```diff
@@ -1529,6 +1535,17 @@ impl TypeChecker {
                         vars
                     ),
                 },
+
+                Constraint::Variable => match self.find_type(a) {
+                    Type::Void => err_type_error!(
+                        self,
+                        span,
+                        TypeError::Exotic,
```

Comment:
As it is right now this error is quite common.

Reply:
But not common in the code ;)